backend/services/inventory/tests/tests/client.py

```diff
@@ -220,70 +225,29 @@ def migrate(self, tenant_id=None, **kwargs):
         return code, stdout, stderr
 
 
-class ApiClient:
-
-    log = logging.getLogger("client.ApiClient")
-
-    def make_api_url(self, path):
-        return os.path.join(
-            self.api_url, path if not path.startswith("/") else path[1:]
-        )
-
-    def setup_swagger(self, host, spec):
-        config = {
-            "also_return_response": True,
-            "validate_responses": True,
-            "validate_requests": False,
-            "validate_swagger_spec": False,
-            "use_models": True,
-        }
-
-        self.http_client = RequestsClient()
-        self.http_client.session.verify = False
-
-        self.client = SwaggerClient.from_spec(
-            load_file(spec), config=config, http_client=self.http_client
-        )
-        self.client.swagger_spec.api_url = self.api_url
-
-    def __init__(self, host, spec):
-        self.setup_swagger(host, spec)
-
-
-class InternalApiClient(ApiClient):
+class InternalApiClient:
     log = logging.getLogger("client.InternalClient")
 
-    def __init__(self, host, spec):
-        self.api_url = "http://%s/api/internal/v1/inventory/" % host
-        super().__init__(host, spec)
-
-    def verify(self, token, uri="/api/management/1.0/auth/verify", method="POST"):
-        if not token.startswith("Bearer "):
-            token = "Bearer " + token
-        return self.client.auth.post_auth_verify(
-            **{
-                "Authorization": token,
-                "X-Forwarded-Uri": uri,
-                "X-Forwarded-Method": method,
-            }
-        ).result()
+    def __init__(self):
+        api_conf = internal_v1.Configuration.get_default_copy()
+        self.client = internal_v1.InternalAPIClient(internal_v1.ApiClient(api_conf))
 
     def DeviceNew(self, **kwargs):
-        return self.client.get_model("DeviceNew")(**kwargs)
+        return internal_v1.DeviceNew(**kwargs)
 
     def Attribute(self, **kwargs):
-        return self.client.get_model("Attribute")(**kwargs)
+        # Return a dict instead of the model to allow flexible value types
+        # (the spec says string, but the backend accepts number, string, array, etc.)
```

Comment:
The spec should be expressible in OAS3 (using oneOf), but I suspect that it might be too complex for the code generator.